_posts/2021-09-07-dd_00003_logic_gates.md

```diff
@@ -0,0 +1,347 @@
+---
+layout: single
+title: Logic Gates
+category: Digital Design
+permalink: /categories/digital_design/logic_gates
+excerpt: "What makes computers think."
+author: Kirthihan Yasotharan
+toc: true
+toc_label: "Content"
+toc_sticky: true
+toc_icon: "align-justify"
+#sidebar:
+#  nav: "getting_started_0"
+header:
+  overlay_color: "#000"
+  overlay_filter: "0.5"
+  overlay_image: /assets/images/digital_design.jpg
+---
+
+<!--
+Topics:
+* Binary numbers / Boolean Algebra
+* Truth tables
+* Gates: and, or, not, xor
+    * introduce xor using complex circuits → use the AND/OR/NOT-XOR representation circuit and make it an exercise "do you notice a special pattern in the final truth table? The output is only on when exactly one input is turned on. And in fact, this behavior turns out to be so important in digital circuits, that there is even a dedicated logic gate for it: the XOR gate (exclusive OR)."
+* Complex circuits
+    * OR = ~(~A & ~B)
+    * boolean equations ("is there a simpler way to describe/represent a complex logic circuit other than a huge truth table?")
+    * + excercises
+
+-->
+
+Last time, we learned the basics of
+[transistors](/categories/digital_design/transistors). Now, it's time to learn
+about **logic gates**. Logic gates can be built using transistors, and are the
+fundamental building blocks of any complex digital circuit. We will split this
+topic into two posts. In this post, we will cover how logic gate work in
+principle. In the post following that, we will see how we can use transistors to
+build logic gates.
+
+Let's get started!
+
+## Logic Operations and Truth Tables
+
+As the name suggests, logic gates perform so-called **logic operations**. A
+logical operation takes some inputs, and combines them in a specific way to
+produce an output. Each input can be either "true" or "false" (or "1", or "0").
+Same holds for the output. For example, the logical "**AND**" operation has two
+inputs, A and B, and one output, Z. The output Z will only be "true" when both A
+_and_ B are "true", hence "and".
+
+The behavior of any logical operation can be respresented using a special table,
+called the **truth table**. The truth table shows for all possible input
+combinations the corresponding output values, thereby fully characterizing the
+function of the particular logic operation. For example, the truth table of the
+AND operation looks like this:
+
+| A   | B   | Z   |
+| --- | --- | --- |
+| 0   | 0   | 0   |
+| 0   | 1   | 0   |
+| 1   | 0   | 0   |
+| 1   | 1   | 1   |
+
+As you can see, the output, Z, is only "on" (1) when _both_ inputs, A and B, are
+"on"; otherwise, the output stays "off" (0). Let's look at the truth table of
+another logic operation, the **OR**:
```

Comment:
Maybe one clear example how to read it. Something like "As you can, the output Z is only on for the last row, where A AND B are on ....

Reply:
Fixed